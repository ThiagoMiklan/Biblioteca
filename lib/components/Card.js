"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var React = _interopRequireWildcard(require("react"));

var _Image = _interopRequireDefault(require("./Image"));

var _Icon = _interopRequireDefault(require("./Icon"));

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  title: "string",
  title_definition: "string",
  definition: "string",
  src: "string",
  itens_footer: "array",
  footer_definition: "string",
  icon_name: "string",
  icon_definition: "string",
  header_definition: "string",
  card_header_definition: "string",
  image_definition: "string",
  card_image_definition: "string",
  icon_onClick: "function",
  image_onClick: "function"
};

var Card = function Card(props) {
  (0, _type_validations.validate)(props, props_obj, "Card");
  var definition = props.definition == undefined ? "card" : "card " + props.definition;
  return /*#__PURE__*/React.createElement("div", {
    className: definition
  }, assembleHeader(props), props.src != undefined && assembleCardImage(props), assembleContent(props), props.itens_footer != undefined && assembleFooter(props));
};

function assembleHeader(props) {
  var header = /*#__PURE__*/React.createElement(React.Fragment, null);

  if (props.custom_header != undefined) {
    header = assembleHeaderCustom(props);
  } else {
    if (props.title != undefined) {
      header = assembleHeaderDefault(props);
    }
  }

  return header;
}

function assembleHeaderDefault(props) {
  var header_definition = props.header_definition == undefined ? "card-header" : "card-header " + props.header_definition;
  var card_header_definition = props.card_header_definition == undefined ? "card-header-icon" : "card-header-icon " + props.card_header_definition;
  var title_definition = props.title_definition == undefined ? "card-header-title" : "card-header-title " + props.title_definition;
  return /*#__PURE__*/React.createElement("header", {
    className: header_definition
  }, /*#__PURE__*/React.createElement("div", {
    className: title_definition
  }, props.title), props.icon_name != undefined && /*#__PURE__*/React.createElement("a", {
    href: "#",
    className: card_header_definition
  }, /*#__PURE__*/React.createElement(_Icon["default"], {
    definition: props.icon_definition,
    icon_name: props.icon_name,
    onClick: props.icon_onClick
  })));
}

function assembleHeaderCustom(props) {
  var header_definition = props.header_definition == undefined ? "card-header" : "card-header " + props.header_definition;
  return /*#__PURE__*/React.createElement("header", {
    className: header_definition
  }, props.custom_header);
}

function assembleCardImage(props) {
  var card_image_definition = props.card_image_definition == undefined ? "card-image" : "card-image " + props.card_image_definition;
  return /*#__PURE__*/React.createElement("div", {
    className: card_image_definition
  }, /*#__PURE__*/React.createElement(_Image["default"], {
    definition: props.image_definition,
    src: props.src,
    onClick: props.image_onClick
  }));
}

function assembleContent(props) {
  return /*#__PURE__*/React.createElement("div", {
    className: "card-content "
  }, props.children);
}

function assembleFooter(props) {
  var footer_definition = props.footer_definition == undefined ? "card-footer" : "card-footer " + props.footer_definition;
  var code = /*#__PURE__*/React.createElement(React.Fragment, null);

  if (props.itens_footer != undefined) {
    code = /*#__PURE__*/React.createElement("footer", {
      className: footer_definition
    }, props.itens_footer.map(function (item) {
      return assembleItemFooter(item);
    }));
  }

  return code;
}

function assembleItemFooter(item) {
  return /*#__PURE__*/React.createElement("p", {
    className: "card-footer-item"
  }, /*#__PURE__*/React.createElement("span", (0, _extends2["default"])({
    onClick: item["onClick"],
    style: {
      cursor: "pointer"
    }
  }, item["custom"]), item["value"]));
}

var _default = Card;
exports["default"] = _default;