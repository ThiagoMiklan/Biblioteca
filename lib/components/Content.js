"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _type_validations = require("../tools/type_validations.js");

var props_obj = {
  definition: 'string'
};

var Content = function Content(props) {
  (0, _type_validations.validate)(props, props_obj, "Content");
  var definition = props.definition == undefined ? "content" : "content " + props.definition;
  return /*#__PURE__*/React.createElement("div", {
    className: definition
  }, props.children);
};

var _default = Content;
exports.default = _default;