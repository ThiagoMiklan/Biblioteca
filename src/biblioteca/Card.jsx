import React from 'react';
import PropTypes from 'prop-types';
import Image from './Image';
import classnames from 'classnames';
import getClassName from '../tools/getClassName';
import Icon from './Icon';


const Card = (props) => {
    // se title for undefined não tem motivo para criar header
    // se src for undefined não tem motivo para criar card image
    // se itens_footer for undefined não tem pq criar footer
	var definition = "card " + classnames(getClassName(props.definition, "Card"));
    return (
        <div className={definition}>
            {assembleHeader(props)}
            {props.src != undefined && assembleCardImage(props)}
            {assembleContent(props)}
            {props.itens_footer != undefined && assembleFooter(props)}
        </div>
    );
}

//  Essa função server para adicionar uma propriedade de customização do header de Card
// Com ela é possível tanto usar o Header Padrão, como informar seu próprio componente para header
// através da props custom.header;
function assembleHeader(props) {
    var header = <></>;
    if (props.custom_header != undefined) {
        header = assembleHeaderCustom(props);
    } else {
        if (props.title != undefined) {
        header = assembleHeaderDefault(props);
        }
    }
    return header;
}

// title: Testando,  icon_name: "fas fa-angle-down", icon_definition: "info", href
// expandido para diversos tipos de valores
function assembleHeaderDefault(props) {
    var header_definition = "card-header " + classnames(getClassName(props.header_definition, "Card"));
    var card_header_definition = "card-header-icon" + classnames(getClassName(props.card_header_definition, "Card"));
    var title_definition = "card-header-title " + classnames(getClassName(props.title_definition, "Card"));

    return <header className={header_definition}>
        <div className={title_definition}>
            {props.title}
        </div>
        {props.icon_name != undefined && <a href="#" className={card_header_definition} >
            <Icon definition={props.icon_definition} icon_name={props.icon_name}></Icon>
        </a>
        }

    </header>
}

function assembleHeaderCustom(props) {
    var header_definition = "card-header " + classnames(getClassName(props.header_definition, "Card"));

    return <header className={header_definition}>
        {props.custom_header}
    </header>
}

// src(props):  image_definition(props): 
// Limite de uma imagem
function assembleCardImage(props) {
    // <Image src="" definition=""></Image>
    var card_image_definition = "card-image " + classnames(getClassName(props.card_image_definition, "Card"));
    return <div className={card_image_definition}>
        <Image definition={props.image_definition} src={props.src}></Image>
    </div>
}

// content
function assembleContent(props) {
    return <div className="card-content ">
        {props.children}
    </div>;
}

// itens_footer(props) footer_definition
// Limitado apenas nessa formado de lista
function assembleFooter(props) {
    var footer_definition = "card-footer " + classnames(getClassName(props.footer_definition, "Card"));
    return <footer className={footer_definition}>
        {props.itens_footer.map(item => assembleItemFooter(item))}
    </footer>
}

function assembleItemFooter(item) {
    return <p className="card-footer-item">
        <span onClick={item["onClick"]} style={{ cursor: "pointer" }}>
            {item["value"]}
        </span>
    </p>
}

Card.propTypes = {
    title: PropTypes.string,
    header_definition: PropTypes.string,
    card_header_definition: PropTypes.string,
    content_definition: PropTypes.string,
    itens_footer: PropTypes.array,
    footer_definition: PropTypes.string,
    image_definition: PropTypes.string,
    src: PropTypes.string,
    icon_definition: PropTypes.string,
    icon_name: PropTypes.string,
    card_image_definition: PropTypes.string,
    custom_header: PropTypes.any,
	definition: PropTypes.string
}



export default Card;