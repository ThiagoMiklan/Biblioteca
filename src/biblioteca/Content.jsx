// @flow
import * as React from 'react';

type Props = {
  definition?: string,
  children?: React.Node
 }

const Content = (props: Props) =>{
    var definition =  (props.definition == undefined) ? "content": "content "+ props.definition;
     return (
        <div class={definition}>
            {props.children}
        </div>
        );
    
}

export default Content;