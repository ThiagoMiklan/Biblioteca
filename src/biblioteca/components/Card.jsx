import {React,useState} from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import getClassName from '../../tools/getClassName';
import { useState } from 'react';
 
const Card = (props) =>{

    return (
        <div className="card">
            {assembleImage()}

        </div>
    );
}

function assembleImage(props){
    return (
            <div className="card-image">  
              {this.props.card_image}
            </div>
            );
}


export default Card;