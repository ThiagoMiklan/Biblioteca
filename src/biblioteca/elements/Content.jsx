import React from 'react';
import getClassName from '../../tools/getClassName';
import className from 'classnames';
import {renderToString} from 'react-dom/server'

class Content extends React.Component {

    render() {
        return (
        <div class="content">
            {assembleList(this.props)}
        </div>
        );
    }
}

 // monta o esqueleto da(s) lista(s) atrav√©s do map e depois monta cada linha de cada lista
 function assembleList(props) {
    var list = props.value;
    
    var list_tags =
        <div className="content">
            {list.map(list_item =>
                <ol className={className(getClassName(list_item["definition"], "Content"))}>
                    {assembleRow(list_item["value"])}
                </ol>
            )}
        </div>

        return list_tags;
}

function assembleRow(value){
    return value.map(item => <li>{item}</li>);
}




export default Content;