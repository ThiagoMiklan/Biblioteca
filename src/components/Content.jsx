// @flow
import * as React from 'react';
import {validate} from '../tools/type_validations.js';

type Props = {
  definition?: string,
  children?: React.Node
 }

const props_obj ={
    definition:  'string'
}

const Content = (props: Props) =>{
    validate(props, props_obj);
    var definition =  (props.definition == undefined) ? "content": "content "+ props.definition;
     return (
        <div class={definition}>
            {props.children}
        </div>
        );
    
}

export default Content;