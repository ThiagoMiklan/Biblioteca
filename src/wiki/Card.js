import React, { useState } from 'react';
import ReactDOM from 'react-dom';
import 'bulma/css/bulma.css';
import Card from './components/Card';
import Column from './components/Column';
import Columns from './components/Columns';
import Content from './components/Content';

const itens_footer = [
  { value: "Save" , onClick: ()=> alert('Clicked')}, { value: "Edit" }, { value: "Delete" }
]

const itens_footer_2 = [
  {value: <span>
          View on <a href="https://twitter.com/codinghorror/status/506010907021828096">Twitter</a>
          </span>},
  {value: <span>
          Share on <a href="#">Facebook</a>
          </span>}
]


function imageClick(){
  alert("Clicou na imagem");
}
ReactDOM.render(
  <Columns definition="is-multiline">
    <Column definition="is-one-quarter">
   <Card definition="has-background-light"
            image_definition="is-4by3"
            src="https://bulma.io/images/placeholders/1280x960.png">

        <div class="media">
          <div class="media-left">
            <figure class="image is-48x48">
              <img src="https://bulma.io/images/placeholders/96x96.png" alt="Placeholder image" />
            </figure>
          </div>
          <div class="media-content">
            <p class="title is-4">John Smith</p>
            <p class="subtitle is-6">@johnsmith</p>
          </div>
        </div>

        <div class="content">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          Phasellus nec iaculis mauris. <a>@bulmaio</a>.
          <a href="#">#css</a> <a href="#">#responsive</a>
          <br />
          <time datetime="2016-1-1">11:09 PM - 1 Jan 2016</time>
        </div>
      </Card>
    </Column>

    <Column definition="is-one-quarter">
     <Card icon_name="fas fa-angle-down" title="Component" itens_footer={itens_footer}>
        <Content>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus nec iaculis mauris.
           <a href="#">@bulmaio</a>. <a href="#">#css</a> <a href="#">#responsive</a>
          <br />
          <time datetime="2016-1-1">11:09 PM - 1 Jan 2016</time>
        </Content>
      </Card>
    </Column>

    <Column definition="is-one-quarter" >
      <Card itens_footer={itens_footer_2}>
        <p class="title">
          “There are two hard things in computer science: cache invalidation, naming things, and off-by-one errors.”
        </p>
        <p class="subtitle">
          Jeff Atwood
        </p>
      </Card>
    </Column>

  </Columns>
,

  document.getElementById("root"))


