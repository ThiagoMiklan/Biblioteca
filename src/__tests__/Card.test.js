import React from 'react';
import Card from '../biblioteca/Card';
import Content from '../biblioteca/Content';
import renderer from 'react-test-renderer';


var itens_footer = [
  { value: "Save" }, { value: "Edit" }, { value: "Delete" }
]

var itens_footer_2 = [
  {value: <span>
          View on <a href="https://twitter.com/codinghorror/status/506010907021828096">Twitter</a>
          </span>},
  {value: <span>
          Share on <a href="#">Facebook</a>
          </span>}
]

test('card 1 with media object',()=>{
    const cardWithMedia = renderer.create(
        <Card definition="background-light"
            image_definition="4by3"
            src="https://bulma.io/images/placeholders/1280x960.png">

        <div class="media">
          <div class="media-left">
            <figure class="image is-48x48">
              <img src="https://bulma.io/images/placeholders/96x96.png" alt="Placeholder image" />
            </figure>
          </div>
          <div class="media-content">
            <p class="title is-4">John Smith</p>
            <p class="subtitle is-6">@johnsmith</p>
          </div>
        </div>

        <div class="content">
          Lorem ipsum dolor sit amet, consectetur adipiscing elit.
          Phasellus nec iaculis mauris. <a>@bulmaio</a>.
          <a href="#">#css</a> <a href="#">#responsive</a>
          <br />
          <time datetime="2016-1-1">11:09 PM - 1 Jan 2016</time>
        </div>
      </Card>
    ).toJSON();

    expect(cardWithMedia).toMatchSnapshot();
})

test('card with content',()=>{
    const cardWithContent = renderer.create(
        <Card icon_name="fas fa-angle-down" title="Component" itens_footer={itens_footer}>
        <Content>
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus nec iaculis mauris.
           <a href="#">@bulmaio</a>. <a href="#">#css</a> <a href="#">#responsive</a>
          <br />
          <time datetime="2016-1-1">11:09 PM - 1 Jan 2016</time>
        </Content>
      </Card>
    ).toJSON();

    expect(cardWithContent).toMatchSnapshot();
})

test('card with title and subtitle',()=>{
    const cardWithTitle = renderer.create(
        <Card itens_footer={itens_footer_2}>
        <p class="title">
          “There are two hard things in computer science: cache invalidation, naming things, and off-by-one errors.”
        </p>
        <p class="subtitle">
          Jeff Atwood
        </p>
      </Card>
    ).toJSON();

    expect(cardWithTitle).toMatchSnapshot();
})
