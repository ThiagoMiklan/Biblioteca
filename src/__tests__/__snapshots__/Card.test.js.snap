// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card 1 with media object 1`] = `
<div
  className="card has-background-light"
>
  <div
    className="card-image "
  >
    <figure
      className="image is-4by3"
    >
      <img
        src="https://bulma.io/images/placeholders/1280x960.png"
      />
    </figure>
  </div>
  <div
    className="card-content "
  >
    <div
      class="media"
    >
      <div
        class="media-left"
      >
        <figure
          class="image is-48x48"
        >
          <img
            alt="Placeholder image"
            src="https://bulma.io/images/placeholders/96x96.png"
          />
        </figure>
      </div>
      <div
        class="media-content"
      >
        <p
          class="title is-4"
        >
          John Smith
        </p>
        <p
          class="subtitle is-6"
        >
          @johnsmith
        </p>
      </div>
    </div>
    <div
      class="content"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus nec iaculis mauris. 
      <a>
        @bulmaio
      </a>
      .
      <a
        href="#"
      >
        #css
      </a>
       
      <a
        href="#"
      >
        #responsive
      </a>
      <br />
      <time
        datetime="2016-1-1"
      >
        11:09 PM - 1 Jan 2016
      </time>
    </div>
  </div>
</div>
`;

exports[`card with content 1`] = `
<div
  className="card "
>
  <header
    className="card-header "
  >
    <div
      className="card-header-title "
    >
      Component
    </div>
    <a
      className="card-header-icon"
      href="#"
    >
      <span />
      <span
        className="icon"
      >
        <i
          aria-hidden="true"
          className="fas fa-angle-down"
        />
      </span>
    </a>
  </header>
  <div
    className="card-content "
  >
    <div
      class="content"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Phasellus nec iaculis mauris.
      <a
        href="#"
      >
        @bulmaio
      </a>
      . 
      <a
        href="#"
      >
        #css
      </a>
       
      <a
        href="#"
      >
        #responsive
      </a>
      <br />
      <time
        datetime="2016-1-1"
      >
        11:09 PM - 1 Jan 2016
      </time>
    </div>
  </div>
  <footer
    className="card-footer "
  >
    <p
      className="card-footer-item"
    >
      <span
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Save
      </span>
    </p>
    <p
      className="card-footer-item"
    >
      <span
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Edit
      </span>
    </p>
    <p
      className="card-footer-item"
    >
      <span
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        Delete
      </span>
    </p>
  </footer>
</div>
`;

exports[`card with title and subtitle 1`] = `
<div
  className="card "
>
  <div
    className="card-content "
  >
    <p
      class="title"
    >
      “There are two hard things in computer science: cache invalidation, naming things, and off-by-one errors.”
    </p>
    <p
      class="subtitle"
    >
      Jeff Atwood
    </p>
  </div>
  <footer
    className="card-footer "
  >
    <p
      className="card-footer-item"
    >
      <span
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        <span>
          View on 
          <a
            href="https://twitter.com/codinghorror/status/506010907021828096"
          >
            Twitter
          </a>
        </span>
      </span>
    </p>
    <p
      className="card-footer-item"
    >
      <span
        style={
          Object {
            "cursor": "pointer",
          }
        }
      >
        <span>
          Share on 
          <a
            href="#"
          >
            Facebook
          </a>
        </span>
      </span>
    </p>
  </footer>
</div>
`;
